### ISSUE TITLE
enhancement: cache(name = {}, options = nil, &block):  new options: if, unless

### ISSUE THREAD

**Issue by eiked** [rails/rails#24014]
I'd like to suggest new options for the CacheHelper#cache method.

:if and :unless

These should work like cache_if (and cache_unless respective) but providing the condition as an option.

aka instead of writing:
    cache_if(foo, key, &block)
I'd like to write:
    cache(key, if:foo, &block)

This might look like syntactic sugar at first and pretty much it is.

My current use case is for temporarily disabling a cache section while debugging production mode.
I could obviously temporarily rewrite the code to use the cache_if method,
but that would change the signature.

But it would feel more natural and less error prone to just attach if:false as an option
(and easily remove that when done with debugging)

This might also be valuable when someone needs conditional caching for whatever reason.

As a bonus, the :if and :unless options should also accept lambdas/procs and evaluate them.
aka
cache(key, unless: ->{foo()}, &block)

**Comment by rafaelfranca** [comment]
Thank you so much for the issue but we don't take feature request on the issue tracker. We recommend you to try to implement the feature and send us a pull request or if you are in doubt if the feature will be accepts ask in Rails Core mailing list for feedback.

I'm fine with this addition though (and I think we may already have a PR for that) so feel free to work on it.

**Comment by eiked** [comment]
Thank you for your fast response.
I fully understand that the bug database should not be polluted with feature requests.
(I got me a red nose last time I suggested an enhancement (while that was already done))

What we're actually missing is a new category to file such enhancement requests,
so that they can be archived and looked up again.

I fully understand, that I'd better implement that myself (and push)
-- but that is not my job of today.
But maybe someone working on that specific part, wants to look that up some day.

Sorry for posting an enhancement again.
Seems you are looking at all new bugs coming in quite immediately,
Big thanks for your immediate response!

I'd believe there should be a category in the bug tracker to file enhancements as well.
(while you? advised that this should be done on the mailing list)

But I believe that enhancement requests and suggestions should also be filed in the bug database.
So that we can comment on it, discuss it, collect opinions and track the implementation state.
(or just close it if it's a bad idea)

I'm not sure if that idea of enhancing the cache method would be a good or bad idea.
But having that in the database would be much more valuable than a post on the list.
Because that way we can work on it, with a process.

----

There must be some way to set up an enhancement category on github.
(or else: write them an enhancement request ;-)

regards
~eike
